hathor/transaction/token_info.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum
+
+
+class TokenInfoVersion(IntEnum):
+    DEPOSIT = 1
+    FEE = 2
+
+
+# used when (de)serializing token information
+@dataclass(slots=True, kw_only=True)
+class TokenInfo:
+    amount: int
+    can_mint: bool
+    can_melt: bool
+    version: TokenInfoVersion | None
```

Comment:
I guess we should have a `NATIVE_TOKEN` type and make this attribute non-optional.